### PR TITLE
Integrated PyCUDD as BDD Solver.

### PR DESCRIPTION
**Pull request by marcogario**
- Added files in patches/ to explain how to install PyCUDD 
 - This also introduces a few fixes to pycudd and the method pycudd.GetDefaultManager()
- Implemented BddSolver as an incremental solver
- Added 'bdd' solver in Factory
- Most tests assumed that all solvers could work on QF_UFLIRA. Explicited added the requirement.
- Added ad-hoc tests

### CODE REVIEW

File: pysmt/solvers/bdd.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    @clear_pending_pop
+    def declare_variable(self, var):
+        self.converter.declare_variable(var)
+
+    @clear_pending_pop
+    def add_assertion(self, formula, named=None):
+        assert_ddmanager(self.ddmanager)
+        bdd_expr = self.converter.convert(formula)
+        _, last_state = self.assertions_stack[-1]
+        new_state = last_state.And(bdd_expr)
```
Comment (mikand) [inline pysmt/solvers/bdd.py]: At stack level n you have the conjunction of all the levels i <= n...
* A comment is needed
* Why doing this computation here and not in the ```solve()``` method?

File: pysmt/solvers/bdd.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+            self.backtrack.append(len(self.assertions_stack))
+
+    @clear_pending_pop
+    def pop(self, levels=1):
+        for _ in xrange(levels):
+            l = self.backtrack.pop()
+            self.assertions_stack = self.assertions_stack[:l]
+
+    def exit(self):
+        raise NotImplementedError
```
Comment (mikand) [inline pysmt/solvers/bdd.py]: Why raising error? Isn't ```pass``` a better option?

File: pysmt/solvers/bdd.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+    def cube_from_var_list(self, var_list):
+        assert_ddmanager(self.ddmanager)
+        indices = pycudd.IntArray(len(var_list))
+        for i, v in enumerate(var_list):
+            indices[i] = self.var2node[v].NodeReadIndex()
+        cube = self.ddmanager.IndicesToCube(indices, len(var_list))
+        return cube
+
+    @deprecated("convert")
```
Comment (mikand) [inline pysmt/solvers/bdd.py]: This is new, so we can avoid to create deprecated code, or is there a specific reason?

File: pysmt/solvers/bdd.py (hunk 1 above)
Comment (marcogario) [inline pysmt/solvers/bdd.py]: Fixed both points in d7d6616

File: pysmt/solvers/bdd.py (hunk 3 above)
Comment (marcogario) [inline pysmt/solvers/bdd.py]: Fixed in feafd06

**Pull request merged**